src/qlever/commands/index.py

```diff
@@ -38,12 +39,95 @@ def additional_arguments(self, subparser) -> None:
                 default=False,
                 help="Overwrite an existing index, think twice before using.")
 
+    # Exception for invalid JSON.
+    class InvalidInputJson(Exception):
+        def __init__(self, error_message, additional_info):
+            self.error_message = error_message
+            self.additional_info = additional_info
+            super().__init__()
+
+    # Helper function to get command line options from JSON.
+    def get_input_options_for_json(self, args) -> str:
+        # Parse the JSON.
+        try:
+            input_specs = json.loads(args.multi_input_json)
+        except Exception as e:
+            raise self.InvalidInputJson(
+                    f"Failed to parse `MULTI_INPUT_JSON` ({e})",
+                    args.multi_input_json)
+            raise Exception("")
+        # Check that it is an array of length at least one.
+        if not isinstance(input_specs, list):
+            raise self.InvalidInputJson(
+                    "`MULTI_INPUT_JSON` must be a JSON array",
+                    args.multi_input_json)
+        if len(input_specs) == 0:
+            raise self.InvalidInputJson(
+                    "`MULTI_INPUT_JSON` must contain at least one element",
+                    args.multi_input_json)
+        # For each of the maps, construct the corresponding command-line
+        # options to the index binary.
+        input_options = []
+        for i, input_spec in enumerate(input_specs):
+            # Check that `input_spec` is a dictionary.
+            if not isinstance(input_spec, dict):
+                raise self.InvalidInputJson(
+                        f"Element {i} in `MULTI_INPUT_JSON` must be a JSON "
+                        "object",
+                        input_spec)
+            # For each `input_spec`, we must have a command.
+            if "cmd" not in input_spec:
+                raise self.InvalidInputJson(
+                        f"Element {i} in `MULTI_INPUT_JSON` must contain a "
+                        "key `cmd`",
+                        input_spec)
+            input_cmd = input_spec["cmd"]
+            # The `format`, `graph`, and `parallel` keys are optional.
+            input_format = input_spec.get("format", args.format)
+            input_graph = input_spec.get("graph", "-")
+            input_parallel = input_spec.get("parallel", "false")
+            # There must not be any other keys.
+            if not all(key in ["cmd", "format", "graph", "parallel"]
+                       for key in input_spec.keys()):
+                raise self.InvalidInputJson(
+                        f"Element {i} in `MULTI_INPUT_JSON` must only contain "
+                        "the keys `format`, `graph`, and `parallel`",
+                        input_spec)
```

Comment:
When rephrasing this condition using sets we can also determine the additional keys.
```suggestion
            extra_keys = input_spec.keys() - {"cmd", "format", "graph", "parallel"}
            if extra_keys:
                raise self.InvalidInputJson(
                        f"Element {i} in `MULTI_INPUT_JSON` must only contain "
                        "the keys `format`, `graph`, and `parallel`. Contains "
                        "extra keys {extra_keys}.",
                        input_spec)
```